From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Wolfgang Bumiller <w.bumiller@proxmox.com>
Date: Mon, 6 Apr 2020 12:16:36 +0200
Subject: [PATCH] PVE: [Config] rbd: block: rbd: disable
 rbd_cache_writethrough_until_flush

Either the cache mode asks for a cache or not. There's no
point in having a "temporary" cache mode. This option AFAIK
was introduced as a hack for ancient virtio drivers. If
anything, we should have a separate option for it. Better
yet, VMs affected by the related issue should simply
explicitly choose writethrough.

Signed-off-by: Wolfgang Bumiller <w.bumiller@proxmox.com>
Signed-off-by: Thomas Lamprecht <t.lamprecht@proxmox.com>
---
 block/rbd.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/block/rbd.c b/block/rbd.c
index 84bb2fa5d7..63f60d41be 100644
--- a/block/rbd.c
+++ b/block/rbd.c
@@ -963,6 +963,8 @@ static int qemu_rbd_connect(rados_t *cluster, rados_ioctx_t *io_ctx,
         rados_conf_set(*cluster, "rbd_cache", "false");
     }
 
+    rados_conf_set(*cluster, "rbd_cache_writethrough_until_flush", "false");
+
     r = rados_connect(*cluster);
     if (r < 0) {
         error_setg_errno(errp, -r, "error connecting");
